acceptance/tests/ticket_1123_facter_with_invalid_locale.rb

```diff
@@ -0,0 +1,9 @@
+test_name "ticket 1123  facter should not crash with invalid locale setting"
+confine :except, :platform => 'windows'
+agents.each do |host|
+  step "set an invalid value for locale and run facter"
+  result = on host, 'LANG=ABCD facter facterversion'
+  fail_test "facter did not warn about the locale " unless
+    result.stderr.include? 'WARN'
```

Comment:
I'd prefer a more precise check, so that we don't accidentally pass if something causes an unrelated warning in the test environment.

How about looking for "locale environment variables were bad"